### DISCUSSION TITLE
Unions in response_models result in incorrect casting

### DISCUSSION THREAD

**Discussion by turettn** [fastapi/fastapi#6423] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Union
from fastapi import FastAPI
from fastapi.testclient import TestClient

APP = FastAPI()
TESTER = TestClient(APP)

@APP.get('/', response_model=Union[int, bool])
def root():
    return True

def test_root():
    resp = TESTER.get("/")
    assert resp.status_code == 200
    assert resp.json() is True

"""
The test fails with:

    def test_root():
        resp = TESTER.get("/")
        assert resp.status_code == 200
>       assert resp.json() is True
E       assert 1 is True
E        +  where 1 = <bound method Response.json of <Response [200]>>()
E        +    where <bound method Response.json of <Response [200]>> = <Response [200]>.json
"""
```

### Description

If response-model is set to `Union[int, bool]`, and return a bool, the result is cast to an `int`.

I expect it to return a bool.

### Operating System

Linux

### FastAPI Version

0.70.0

### Python Version

3.7.10

**Comment by uriyyo** [comment, marked as answer]
`pydantic` tries to convert values to destination type when it's possible. 

The solution will be to use `pydantic` `StrictInt` and `StrictBool`:
```py
from typing import Union
from fastapi import FastAPI
from fastapi.testclient import TestClient
from pydantic import StrictInt, StrictBool

APP = FastAPI()
TESTER = TestClient(APP)


@APP.get('/', response_model=Union[StrictInt, StrictBool])
def root():
    return 1


def test_root():
    resp = TESTER.get("/")
    assert resp.status_code == 200
    assert resp.json() is True
```

More info about `pydantic`:
data conversion - https://pydantic-docs.helpmanual.io/usage/models/#data-conversion
strict types - https://pydantic-docs.helpmanual.io/usage/types/#strict-types

**Comment by turettn** [comment]
@uriyyo Thank you!  I tried to search through the pydantic docs but couldn't find this.  That's perfect.
